File: core/src/ru/myitschool/dcrawler/skills/action/ComplexAttackAction.java
package ru.myitschool.dcrawler.skills.action;

import ru.myitschool.dcrawler.math.MathAction;
import ru.myitschool.dcrawler.skills.Skill;

public class ComplexAttackAction extends ActionAdapter {

    private MathAction successAction;
    private MathAction critSuccessAction;
    private MathAction failAction;
    private MathAction critFailAction;

    public ComplexAttackAction(Skill skill, MathAction successAction, MathAction critSuccessAction, MathAction failAction, MathAction critFailAction) {
        super(skill);
        this.successAction = successAction;
        this.critSuccessAction = critSuccessAction;
        this.failAction = failAction;
        this.critFailAction = critFailAction;
    }

    @Override
    protected MathAction successDamage() {
        return successAction;
    }

    @Override
    protected MathAction critSuccessDamage(int standardDamage) {
        return critSuccessAction;
    }

    @Override
    protected MathAction failDamage() {
        return failAction;
    }

    @Override
    protected MathAction critFailDamage(int standardDamage) {
        return critFailAction;
    }
}

File: core/src/com/wolg_vlad/dcrawler/entities/skills/action/ComplexAttackAction.kt
package com.wolg_vlad.dcrawler.entities.skills.action

import com.wolg_vlad.dcrawler.entities.skills.Skill
import com.wolg_vlad.dcrawler.math.MathAction

class ComplexAttackAction(skill: Skill, private val successAction: MathAction, private val critSuccessAction: MathAction, private val failAction: MathAction?, private val critFailAction: MathAction?) : ActionAdapter(skill) {
    override fun successDamage(): MathAction? {
        return successAction
    }

    override fun critSuccessDamage(standardDamage: Int): MathAction? {
        return critSuccessAction
    }

    override fun failDamage(): MathAction? {
        return failAction
    }

    override fun critFailDamage(standardDamage: Int): MathAction? {
        return critFailAction
    }

}